lambdas/question/src/test/java/uk.gov.di.ipv.cri.kbv.api.handler/QuestionHandlerTest.java

```diff
@@ -0,0 +1,197 @@
+
+@ExtendWith(MockitoExtension.class)
+public class QuestionHandlerTest {
+
+    private QuestionHandler questionHandler;
+    @Mock private ObjectMapper mockObjectMapper;
+    @Mock private StorageService mockStorageService;
+    @Mock private ExperianService mockExperianService;
+
+    @Mock private Appender<ILoggingEvent> appender;
+
+    @BeforeEach
+    void setUp() {
+        AWSXRay.beginSegment("handleRequest");
+        Logger logger = (Logger) LoggerFactory.getLogger(QuestionHandler.class);
+        logger.addAppender(appender);
+        questionHandler =
+                new QuestionHandler(mockObjectMapper, mockStorageService, mockExperianService);
+    }
+
+    @AfterEach
+    void tearDown() {
+        AWSXRay.endSegment();
+    }
+
+    @Test
+    void shouldReturn200OkWhen1stCalledAndReturn1stUnAnsweredQuestion()
+            throws IOException, InterruptedException {
+        APIGatewayProxyRequestEvent input = mock(APIGatewayProxyRequestEvent.class);
+        Map<String, String> sessionHeader = Map.of(HEADER_SESSION_ID, "new-session-id");
+
+        Context contextMock = mock(Context.class);
+        KBVSessionItem kbvSessionItemMock = mock(KBVSessionItem.class);
+        PersonIdentity personIdentityMock = mock(PersonIdentity.class);
+        QuestionState questionStateMock = mock(QuestionState.class);
+
+        when(input.getHeaders()).thenReturn(sessionHeader);
+        when(mockStorageService.getSessionId(sessionHeader.get(HEADER_SESSION_ID)))
+                .thenReturn(Optional.ofNullable(kbvSessionItemMock));
+        when(mockObjectMapper.readValue(
+                        kbvSessionItemMock.getUserAttributes(), PersonIdentity.class))
+                .thenReturn(personIdentityMock);
+        when(mockObjectMapper.readValue(kbvSessionItemMock.getQuestionState(), QuestionState.class))
+                .thenReturn(questionStateMock);
+        when(mockObjectMapper.writeValueAsString(personIdentityMock)).thenReturn("person-identity");
+
+        QuestionsResponse questionsResponseMock = mock(QuestionsResponse.class);
+        when(mockExperianService.getQuestions("person-identity")).thenReturn(questionsResponseMock);
+        when(questionStateMock.setQuestionsResponse(questionsResponseMock)).thenReturn(true);
+        String state = "question-state";
+        when(mockObjectMapper.writeValueAsString(questionStateMock)).thenReturn(state);
+        Control controlMock = mock(Control.class);
+        when(questionStateMock.getControl()).thenReturn(controlMock);
+        when(controlMock.getAuthRefNo()).thenReturn("auth-ref-no");
+        when(controlMock.getURN()).thenReturn("ipv-session-id");
+        doNothing()
+                .when(mockStorageService)
+                .update(
+                        sessionHeader.get(HEADER_SESSION_ID),
+                        state,
+                        "auth-ref-no",
+                        "ipv-session-id");
```

Comment:
same here

Reply:
Try and address some of the 20 code smells highlighted by SonarCloud

Reply:
Addressed a few from the Major and critical code smells highlighted by SonarCloud